events/importer/enkora.py

```diff
@@ -1068,29 +1390,58 @@ def _handle_course(self, course: dict) -> dict:
 
         # Description:
         if course["description_long"]:
-            description = course["description_long"]
+            description = course["description_long"].strip()
         elif course["description"]:
-            description = course["description"]
+            description = course["description"].strip()
         elif course["reservation_event_group_name"]:
-            description = course["reservation_event_group_name"]
+            description = course["reservation_event_group_name"].strip()
 
         # Location:
-        location_id, location_extra_info = self.convert_location(course)
+        location_data, location_extra_info = self.convert_location(course)
+        location_extra_info = {
+            "fi": "Kurssin lisätiedot, Liikuntaluuri: +358 9 310 32623"
+        }
 
         # Keywords
-        keywords = self.convert_keywords(course)
+        location_kwids, service_kwids = self.convert_keywords(course)
 
         # Audience
         (
             language,
-            audience_keywords,
-            sport_keywords,
+            audience_kwids,
+            sport_kwids,
             audience_min_age,
             audience_max_age,
-        ) = self.convert_audience(course, description)
-        keywords.extend(sport_keywords)
+            liikuntakauppa_fi_link,
+        ) = self.convert_audience(course)
         in_language = [self.languages[language]]
 
+        kw_ids = (
+            self.ALL_COURSES_KEYWORDS | location_kwids | service_kwids | sport_kwids
+        )
+        keywords = self.keyword_ids_into_keywords(kw_ids)
+        audience_keywords = self.keyword_ids_into_keywords(audience_kwids)
+
+        keyword_inaccuracy = []
+        keyword_inaccuracy += ["location"] if not location_kwids else []
+        keyword_inaccuracy += ["service"] if not service_kwids else []
+        keyword_inaccuracy += ["sport"] if not sport_kwids else []
+        keyword_inaccuracy += ["audience"] if not audience_kwids else []
+        keyword_accuracy_perc = (1 - len(keyword_inaccuracy) / 4) * 100
+        if keyword_accuracy_perc < 50.0:
+            logging.warning(
+                "Enkora event ID {0} '{1}': Poor mapping ({2:.0f}%) into keywords! "
+                "Suggest improving mapping on: {3}".format(
+                    course["reservation_event_group_id"],
+                    course["reservation_event_group_name"].strip(),
+                    keyword_accuracy_perc,
+                    ", ".join(keyword_inaccuracy),
+                )
+            )
```

Comment:
Change: Probably easier to understand without negations... and can also use percentages where it's relevant:
```suggestion
        keyword_types = []
        keyword_types += ["location"] if location_kwids else []
        keyword_types += ["service"] if service_kwids else []
        keyword_types += ["sport"] if sport_kwids else []
        keyword_types += ["audience"] if audience_kwids else []
        # More keyword types = more accurate
        keyword_accuracy = len(keyword_types) / 4
        if keyword_accuracy < 0.5:
            logging.warning(
                "Enkora event ID {0} '{1}': Poor mapping ({2:.0f}%) into keywords! "
                "Suggest improving mapping on: {3}".format(
                    course["reservation_event_group_id"],
                    course["reservation_event_group_name"].strip(),
                    keyword_accuracy * 100,
                    ", ".join(keyword_types),
                )
            )
```